### PR TITLE
Changes to support ember > 2.1

### PR DESCRIPTION
**Pull request by hyderali**
Uses getOwner method if available to get container and registry is referred from `container._registry` or `container.registry` based on the availability.

### CODE REVIEW

**Comment by hyderali** [comment]
@MiguelMadero , with these changes, the correct route from the package is not picked from the container. I've tracked down to this [line](https://github.com/emberjs/ember.js/blob/v2.7.0/packages/container/lib/registry.js#L846) which returns `null` instead of the factory.  
As a result, the `route:basic` factory is returned instead of the `route:package1` factory.  
I'm digging further. Any help from your side will also be highly useful.  

I'll update here with my latest findings if any.

File: addon/lazy-router.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import Ember from 'ember';
 
-const get = Ember.get;
+const { get, getOwner } = Ember;
```
Comment (MiguelMadero) [inline addon/lazy-router.js]: Should we add https://github.com/rwjblue/ember-getowner-polyfill?

File: addon/lazy-router.js [hunk 2]
```diff
@@ -39,9 +39,29 @@ export default Ember.Router.extend({
       return handler;
     };
   },
+  _getContainer: function() {
+    var hasGetOwner = typeof getOwner === "function";
+    var container = hasGetOwner? getOwner(this) : this.container;
```
Comment (MiguelMadero) [inline addon/lazy-router.js]: Check if `getOwner(this);` has a property called based or parent. we might need that to get the root owner, that might be why the package1 factory isn't there... just a hint from memory, I might be mixing things.

File: addon/lazy-router.js (hunk 1 above)
Comment (hyderali) [inline addon/lazy-router.js]: I don't think so. For apps using older Ember versions, we can continue accessing the container directly.

File: addon/lazy-router.js (hunk 2 above)
Comment (hyderali) [inline addon/lazy-router.js]: I'll check this and update.

Comment (hyderali) [inline addon/lazy-router.js]: @MiguelMadero , Yes, you are correct. `getOwner(this)` has a property called `base` and the factory has to be registered in this base.
I've made those changes and the correct route from the package is not from the container.

I think, this branch is good to go now.

**Comment by MiguelMadero** [comment]
Nice! thanks for fixing that. To make sure we don't break that, can you remove the now supported scenarios from ALLOWED_FAILURES in `travis.yml`? 

Also, if you run `npm test` you will see the list of failures, we need to update the "approved" files to get those tests to pass. 

I can help with any of this, just let me know if you need a hand or need me to take over from here. 

Thanks for the fix!

**Comment by hyderali** [comment]
I've changed the approval files. Tests seems to pass in my dev machine 😲 .
I'm glad that if you can take over from here.

**Comment by MiguelMadero** [comment]
> 😲 .

Sorry about that. I have to figure out the exact problem, it seems like there is a lib/addon or something that leads to a different identity. Since not all dependencies are pinned, depending on when you npm install the identity might be different than what Jenkins generates. This is awful and I need to make this part easier. 

I'll have a look.

**Comment by hyderali** [comment]
Thanks.

**Comment by MiguelMadero** [comment]
@hyderali I have a fix for the tests, check 10c47dde9aaf65f775ace4922e286f79ddfc539b 
It was based on your branch, but I noticed you just rebased. Let me know when this is ready and I can rebase on top of your branch or feel free to cherry-pick the commit above with the test fixes.

**Comment by hyderali** [comment]
Ahh !
I thought squashing the commits will be a better one !
Cherry picked your commit and all is well now 😄 .

**Comment by MiguelMadero** [comment]
This is awesome, thanks!

**Comment by MiguelMadero** [comment]
Published as 0.2.7

**Pull request merged**